x/ccv/types/denom_helpers.go

```diff
@@ -0,0 +1,188 @@
+package types
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	transfertypes "github.com/cosmos/ibc-go/v10/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v10/modules/core/04-channel/types"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cmtbytes "github.com/cometbft/cometbft/libs/bytes"
+	cmttypes "github.com/cometbft/cometbft/types"
+)
+
+// The code in this file is removed in ibc v9. It is copied from ibc v8 to here in order to support the migration to v9
```

Comment:
We should speak to the IBC team and figure out why this was pushed downstream, since most IBC-implementing applications seam to use this